cdap-common/src/main/java/co/cask/cdap/common/zookeeper/ZKExtOperations.java

```diff
@@ -148,7 +158,7 @@
   }
 
   /**
-   * Update the content of the given node. If the node doesn't exists, it will try to create the node. If the node
+   * Update the content of the given node. If the node doesn't exist, it will try to create the node. If the node
```

Comment:
Only if ZK is not available. However, in most algorithms that involves using ZK, you cannot stop without a known state change has been done on ZK, that's why cancel is not supported.